gwcs/api.py

```diff
@@ -183,20 +182,30 @@ def pixel_to_world(self, *pixel_arrays):
         """
         Convert pixel values to world coordinates.
         """
+        if self.forward_transform.uses_quantity and not isinstance(pixel_arrays[0], u.Quantity):
+            pixel_arrays = [pixel * self.input_frame.unit[i]
+                            for i, pixel in enumerate(pixel_arrays)]
         return self(*pixel_arrays, with_units=True)
 
     def array_index_to_world(self, *index_arrays):
         """
         Convert array indices to world coordinates (represented by Astropy
         objects).
         """
-        return self(*(index_arrays[::-1]), with_units=True)
+        pixel_arrays = index_arrays[::-1]
+        if self.forward_transform.uses_quantity and not isinstance(pixel_arrays[0], u.Quantity):
+            pixel_arrays = [pixel * self.input_frame.unit[i]
+                            for i, pixel in enumerate(pixel_arrays)]
+        return self(*pixel_arrays, with_units=True)
 
     def world_to_pixel(self, *world_objects):
```

Comment:
This is the high-level API right?

Why can't we return Quantities here if that's what the model spits out?

Reply:
`astropy.wcs.world_to_pixel` doesn't return Quantities.  I would expect/want consistency from the unified WCS API.  For `astropy.wcs` (FITS), it would have to always return pixels with or without Quantities (or would require a new keyword option to specify the output type).

Reply:
Yeah I suppose consistency seems important. I do hate that `astropy.wcs` is still lowest common denominator everywhere, given how crufty it is.

Reply:
Well, it's essentially FITS WCS, so that's no surprise. :smile: